test/micro/org/openjdk/bench/java/lang/invoke/MethodHandleProxiesAsIFInstanceCall.java

```diff
@@ -0,0 +1,186 @@
+    private Doable lambda;
+
+    static {
+        try {
+            constantTarget = LOOKUP.findStatic(MethodHandleProxiesAsIFInstanceCall.class, "doWork", MT_int_int);
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+        constantDoable = new Doable() {
+            @Override
+            public int doWork(int i) {
+                return MethodHandleProxiesAsIFInstanceCall.doWork(i);
+            }
+        };
+        constantHandle = new Doable() {
+            @Override
+            public int doWork(int i) {
+                try {
+                    return (int) constantTarget.invokeExact((int) i);
+                } catch (Error | RuntimeException e) {
+                    throw e;
+                } catch (Throwable e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        constantInterfaceInstance = MethodHandleProxies.asInterfaceInstance(Doable.class, constantTarget);
+        constantLambda = MethodHandleProxiesAsIFInstanceCall::doWork;
+    }
+
+    @Setup
+    public void setup() throws Throwable {
+        target = LOOKUP.findStatic(MethodHandleProxiesAsIFInstanceCall.class, "doWork", MT_int_int);
+        doable = new Doable() {
+            @Override
+            public int doWork(int i) {
+                return MethodHandleProxiesAsIFInstanceCall.doWork(i);
+            }
+        };
+        handle = new Doable() {
+            @Override
+            public int doWork(int i) {
+                try {
+                    return (int) target.invokeExact((int) i);
+                } catch (Error | RuntimeException e) {
+                    throw e;
+                } catch (Throwable e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        interfaceInstance = MethodHandleProxies.asInterfaceInstance(Doable.class, target);
+        lambda = (Doable) LambdaMetafactory.metafactory(LOOKUP, "doWork", MT_Doable, MT_int_int, target, MT_int_int).getTarget().invokeExact();
+    }
+
+    @Benchmark
+    public void direct() {
+        i = doWork(i);
+    }
+
+    @Benchmark
+    public void callDoable() {
+        i = doable.doWork(i);
+    }
+
+    @Benchmark
+    public void callHandle() {
+        i = handle.doWork(i);
+    }
+
+    @Benchmark
+    public void callInterfaceInstance() {
+        i = interfaceInstance.doWork(i);
+    }
+
+    @Benchmark
+    public void callLambda() {
+        i = lambda.doWork(i);
+    }
+
+    @Benchmark
+    public void constantDoable() {
+        i = constantDoable.doWork(i);
+    }
+
+    @Benchmark
+    public void constantHandle() {
+        i = constantHandle.doWork(i);
+    }
+
+    @Benchmark
+    public void constantInterfaceInstance() {
+        i = constantInterfaceInstance.doWork(i);
+    }
+
+    @Benchmark
+    public void constantLambda() {
+        i = constantLambda.doWork(i);
+    }
```

Comment:
I think setting the result into an instance field like this can work, but it's imo better to let JMH handle it. So, these methods should just return the value instead of writing it to the `i` field.

Reply:
That too. But I'm talking about the value that is assigned to the field after calling `doWork`. That value should be 'consumed' as well. IMO it's better to do that the conventional way, either by returning the value (or using a Blackhole). i.e. let JMH handle it.